src/DynamicData/Cache/ObservableCache.cs

```diff
@@ -240,4 +322,103 @@ private void InvokePreview(ChangeSet<TObject, TKey> changes)
             }
         }
     }
+
+    private void InvokeCountNext()
+    {
+        lock (_locker)
+        {
+            if (_countChanged.IsValueCreated)
+            {
+                _countChanged.Value.OnNext(_readerWriter.Count);
+            }
+        }
+    }
+
+    private void ResumeCount()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Count without Suspend Count instance");
+            _suspensionTracker.Value.ResumeCount();
+        }
+    }
+
+    private void ResumeNotifications()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Notifications without Suspend Count instance");
+            _suspensionTracker.Value.ResumeNotifications();
+        }
+    }
+
+    private class SuspensionTracker(Action<ChangeSet<TObject, TKey>> onResumeNotifications, Action onResumeCount) : IDisposable
+    {
+        private readonly BehaviorSubject<bool> _notifySuspendSubject = new(false);
+
+        private readonly Action<ChangeSet<TObject, TKey>> _onResumeNotifications = onResumeNotifications;
+
+        private readonly Action _onResumeCount = onResumeCount;
+
+        private IEnumerable<Change<TObject, TKey>> _pending = Enumerable.Empty<Change<TObject, TKey>>();
+
+        private int _countSuspendCount;
+
+        private int _notifySuspendCount;
+
+        public bool CountSuspended => _countSuspendCount > 0;
+
+        public bool NotificationsSuspended => _notifySuspendCount > 0;
+
+        public IObservable<bool> NotificationsSuspendedStatus => _notifySuspendSubject;
```

Comment:
Suggest `AreNotificationsSuspended` for the name, which by no coincidence collides with the name I suggested for the plain flag above. That suggests that the two ought to be combined as a `BehaviorSubject<T>` or a read-only equivalent.

Unfortunately, I don't think such a thing (a read-only equivalent of a `BehaviorSubject<T>`) exists, but since this is a private class only, I don't think there's any real danger in just exposing the subject directly.

Hit me up in Slack if all this isn't coming across clearly.

Reply:
I don't want to expose the Subject directly.  If I change it to Subject<int>, then I can expose `_subject.Select(x => x > 0)`.  I guess that doesn't have to allocate if I label it static.

Reply:
I took another stab at the name.  I'd rather not expose the Subject directly, even in an internal class, I don't like exposing implementation details if it can be avoided.

Reply:
> I don't want to expose the Subject directly. If I change it to Subject, then I can expose `_subject.Select(x => x > 0)`. I guess that doesn't have to allocate if I label it static.

If you were gonna do this, the best way would be to have the transformed observable itself stored and reused on the object instance, rather than having every call to the property invoke `.Select()` again, and create a new `IObservable<T>` instance.

I.E.

```cs
public SuspensionTracker()
{
    ...
    _areNotificationsSuspended = _subject.Select(...);
   ...
}

...

public IObservable<bool> AreNotificationsSuspended
    => _areNotificationsSuspended;
```